sw/host/tests/manuf/manuf_cp_device_info_flash_wr/src/main.rs

```diff
@@ -45,22 +55,64 @@ fn manuf_cp_device_info_flash_wr(opts: &Opts, transport: &TransportWrapper) -> R
 
     // Reset and halt the CPU to ensure we are in a known state, and clear out any ROM messages
     // printed over the console.
-    jtag.reset(false)?;
+    jtag.reset(/*run=*/ false)?;
     let uart = transport.uart("console")?;
     uart.clear_rx_buffer()?;
 
-    // Load the SRAM program that contains the test code.
+    // Load and execute the SRAM program that contains the test code.
     match opts
         .sram_program
-        .load_and_execute(&jtag, ExecutionMode::Jump)?
+        .load_and_execute(&jtag, ExecutionMode::Call(opts.timeout))?
     {
         ExecutionResult::CallReturned => {
             log::info!("SRAM program loaded and executed successfully.")
         }
         res => log::info!("SRAM program execution failed: {:?}.", res),
     }
 
-    // Wait for test status pass over the UART.
+    // Once the SRAM program has printed a message over the console, we can continue with a LC
+    // transition initiated on the host side.
+    let _ = UartConsole::wait_for(
+        &*uart,
+        r"Done. Perform an LC transition and run flash stage.",
+        opts.timeout,
+    )?;
+
+    // Reset and halt the CPU to ensure we are in a known state again, clear out any ROM
+    // messages printed over the console, anw wwitch to the LC TAP to perform an LC
```

Comment:
and switch [...] and perform a transition to `PROD`. Other comments need to be updated as well.  Maybe it makes sense to support `PROD`, `DEV` and `PROD_END`, and skip the flash check in the .c side if the dut is in `DEV` state.

Reply:
fixed.